File: wk7/java/validPalindrome.java
// problem link: https://leetcode.com/problems/valid-palindrome/description/

class Solution {
    public boolean isPalindrome(String s) {
        int fro = 0, bac = s.length() - 1;
        while (fro < bac) {
            if (!Character.isLetterOrDigit(s.charAt(fro)))
                fro++;
            else if (!Character.isLetterOrDigit(s.charAt(bac)))
                bac--;
            else {
                if (Character.toLowerCase(s.charAt(fro)) != Character.toLowerCase(s.charAt(bac)))
                    return false;
                fro++;
                bac--;
            }
        }
        return true;
    }
}

File: wk7/cpp/validPalindrome.cpp
// problem link: https://leetcode.com/problems/valid-palindrome/description/

class Solution {
public:
    bool isPalindrome(string s) {
        int f = 0, b = s.size() - 1;
        while(f < b) {
            if(!isalpha(s[f]) && !isdigit(s[f]))
                f++;
            else if(!isalpha(s[b]) && !isdigit(s[b]))
                b--;
            else {
                if(tolower(s[f]) != tolower(s[b]))
                    return false;
                f++;
                b--;
            }
        }
        return true;
    }
};